### PR TITLE
[fix] Use metrics decorator around MetricStorage

### PR DESCRIPTION
**Pull request by yurishkuro**
## Which problem is this PR solving?
- The SPM Troubleshooting docs were referring to metrics that are supposed to be emitted about MetricStore operations, but no such metrics were present in v2. It turns out the metrics decorator was not applied to MetricStore.

## Description of the changes
- Apply decorator
- Rename decorator package to `metricstoremetrics`
- Use concrete v3.0.0. version of Promethus in the docker-compose file

## How was this change tested?
```
$ curl -s http://localhost:8888/metrics | grep -v '^#' | grep get_call_rates | head -3
metricstore_latency_bucket{operation="get_call_rates",result="ok",service_instance_id="463e3888-a9f8-4b10-8fd7-62aa0ccd6fe3",service_name="jaeger",service_version="v2.0.0",le="0"} 0
metricstore_latency_bucket{operation="get_call_rates",result="ok",service_instance_id="463e3888-a9f8-4b10-8fd7-62aa0ccd6fe3",service_name="jaeger",service_version="v2.0.0",le="5"} 2
metricstore_latency_bucket{operation="get_call_rates",result="ok",service_instance_id="463e3888-a9f8-4b10-8fd7-62aa0ccd6fe3",service_name="jaeger",service_version="v2.0.0",le="10"} 2
```

### CODE REVIEW

File: docker-compose/monitor/README.md [hunk 1]
```diff
@@ -81,7 +81,7 @@ docker compose -f docker-compose-v2.yml up
 **Tips:**
 - Let the application run for a couple of minutes to ensure there is enough time series data to plot in the dashboard.
 - Navigate to Jaeger UI at http://localhost:16686/ and inspect the Monitor tab. Select `redis` service from the dropdown to see more than one endpoint.
-- To visualize the raw metrics stored on the Prometheus server (for debugging and local development use cases), use the built-in Prometheus UI at http://localhost:9090/query. For example, http://localhost:9090/query?g0.expr=traces_span_metrics_calls_total&g0.tab=0&g0.range_input=5m
+- To visualize the raw metrics stored on the Prometheus server (for debugging and local development use cases), use the built-in Prometheus UI at http://localhost:9090/query. For example, http://localhost:9090/graph?g0.expr=traces_span_metrics_calls_total&g0.tab=0&g0.range_input=5m
```
Comment (mahadzaryab1) [inline docker-compose/monitor/README.md:84]: does the `graph` endpoint still work? I thought it was replaced by the `query` endpoint.

**Review by mahadzaryab1 — COMMENTED** [review]
Just a couple of questions - i also realized I forgot to rename [this](https://github.com/jaegertracing/jaeger/blob/main/storage/spanstore/spanstoremetrics/decorator_test.go) file (it should be `read_metrics_test.go`). Can that be fixed in this PR as well? Otherwise I can open a separate PR for it.

File: docker-compose/monitor/README.md (hunk 1 above)
Comment (yurishkuro) [reply docker-compose/monitor/README.md:84]: Query didn't work for me with the version of Prometheus that is invoked from docker compose file. Did you run a different version? 3.x has been released.

Comment (mahadzaryab1) [reply docker-compose/monitor/README.md:84]: Hm that's weird - I remember using `graph` was causing the CI to fail which was fixed in https://github.com/jaegertracing/jaeger/pull/6217

**Review by mahadzaryab1 — APPROVED** [review]

**Pull request merged**